src/lib/secretstore.js

```diff
@@ -14,17 +14,19 @@ class SecretStore {
         if(!options.awsOpts.region)
             throw new Error('AWS Region must be defined');
 
-        if(options.cacheBuster) {
-            // check the cache buster key every minute
-            const cacheBusterKey = options.cacheBuster;
-            setInterval(() => {
-                this.getSecret({name: cacheBusterKey}).then(secret => this.cacheRefreshTime = Number(secret));
-            }, 60000);
-            delete options.cacheBuster;
-        }
+        const cacheBuster = options.cacheBuster;
+        delete options.cacheBuster;
         this.options = options;
         this.store = new Credstash(options);
         this.cache = {};
+
+        // check the cache buster key every minute
+        if(cacheBuster) {
+            const cacheBusterKey = secretGen(cacheBuster);
+            setInterval(() => {
```

Comment:
This hides errors with refetching errors. We should opt for keeping a local store of the secret and upon fetching it spin off a new _get_ when it's past it's TTL.

Reply:
We opted for this instead of a ttl so we could just invalidate all the cached secrets at once if needed. The only swallowed exception would be if we couldn't get the cacheBuster secret, which is ok.

Reply:
Should at least have some sort of notification for when the refresh is failing. Currently this masks any errors.

Reply:
I think it throws an unhandled exception right now from the store.getSecret. Should I log that? I figured the exception in the logs would be good enough.

Reply:
Because it's called in a SetInterval it would be unhandled, but it wouldn't cause an error to be thrown in any real sense since it broke out of the chain.

[Here is a proof of what I'm talking about](https://github.com/asap/reimagined-spoon)

Reply:
(Updated my comment with the correct link)

Reply:
PR addressing this: https://github.com/Asymmetrik/yadda-secret/pull/9